### PR TITLE
Display mod.io tags

### LINKED ISSUE #4: Display mod's mod.io tags

### PR DESCRIPTION
**Pull request by jieyouxu**
Closes #4.

Screenshot 2023-07-29 074220

Also made some drive-by changes:

- Bump dependencies:
    - One or more of the dependencies used a version of `proc-macro2` that is outdated causing compilation to fail on nightly.
- Specify rust toolchain to be stable channel.
- Update and format `README.md`.
- Format with default rustfmt settings.
- Enlarge initial window size a bit.

### CODE REVIEW

**Comment by trumank** [comment]
Awesome! I'll elaborate on some of my thoughts for the feature. Not all of them are necessary to merge, though I would greatly appreciate if you were able to work on them. :)

I think this screenshot confirms my suspicion that some processing/collapsing of tags should be done before displaying:

image

I think it would be preferable if the tags were right aligned and in a specific order like so:

    | mod name                    [QoL]? [gameplay]? [audio]? [visual]? [required status] [approval status] |

`[required status]` and `[approval status]` would always be present for all modio mods

The game version tags are mostly meaningless so probably not necessary to give them any screen space. The full sorted tag list could be hidden in some hover text for the mod if desired.

Right aligning is preferred to keep the data roughly tabular. I think it's doable, but just a heads up: the scroll box has been giving me some trouble (try shrinking the window so the mods overflow and watch the scrollbar move to the right slightly). 

It would also be great if the search feature highlighted tags as well:
image

**Comment by jieyouxu** [comment]
Good call on the processing being required. That is indeed very noisy and hard to navigate. I will investigate adding some processing for the tags.

**Comment by jieyouxu** [comment]
> Right aligning is preferred to keep the data roughly tabular. I think it's doable, but just a heads up: the scroll box has been giving me some trouble (try shrinking the window so the mods overflow and watch the scrollbar move to the right slightly).

I have no clue how to right-align these labels. Each row in this draggable UI is a `ui.horizontal`, which AFAIK doesn't allow horizontal alignment adjustments? I've tried to use columns, but that didn't work very well either. Any ideas?

**Comment by trumank** [comment]
You can wrap the tags in `ui.with_layout(egui::Layout::right_to_left(egui::Align::Center), |ui| { ... });`

**Comment by jieyouxu** [comment]
I'll see if I can make the labels also searchable tomorrow.

**Comment by jieyouxu** [comment]
Made mod.io tags searchable.

**Comment by jieyouxu** [comment]
Made approval status tags colored to help them stand out.

**Comment by jieyouxu** [comment]
Reordered flags so approval status and required status are on the right instead.

File: README.md [hunk 1]
```diff
@@ -1,14 +1,17 @@
-# drg-mod-integration
-3rd party mod integration tool to download and integrate mods completely
-externally of the game. This enables more stable mod usage as well as offline
-mod usage. Works for both Steam and Microsoft Store versions.
+# Deep Rock Galactic Mod Integration
```
Comment (trumank) [inline README.md:1]: I would prefer the title remains the crate name and not have the full game name in the title. Even 'drg' is more than I'd like but naming things is hard...

**Pull request merged**